### ISSUE TITLE
[Xcode 15-Beta] - The signature of “JRE.xcframework” cannot be verified

### ISSUE THREAD

**Issue by ferencIOS** [google/j2objc#2101]
Hi all.

New WWDC new problems!

When I try to run the app that contains the JRE.xcframework, the new Xcode shows me this error:
`Codesign verification failed`

They must surely have introduced some new flag that is not known.

(I'm not alone: https://developer.apple.com/forums/thread/730955)

Thanks for your help!

**Comment by tomball** [comment]
Are you building j2objc as part of your app build, or using a pre-built release? We're going to release a new version once the JRE update is finished, as the current version is likely out-of-date.

**Comment by ferencIOS** [comment]
Basically I followed what I had answered some time ago [here](https://github.com/google/j2objc/issues/1933#issuecomment-1338972728)!

Then I integrated the xcframework into the project.

They probably added some new flag for:
```
xcodebuild clean archive ...
```

In fact it works with Xcode 14 and it doesn't work with Xcode 15

**Comment by ferencIOS** [comment]
Following the Apple WWDC:

https://developer.apple.com/documentation/Xcode/verifying-the-origin-of-your-xcframeworks
or
https://developer.apple.com/videos/play/wwdc2023/10061/

which simplifying
```
xcodebuild -create-xcframework [...] -output <YOUR.xcframework>
codesign --timestamp -v --sign "<YOUR CERTIFICATE (ABCXYZ)>" "<YOUR.xcframework>"
```

**Comment by dsanghan** [comment]
Yup seeing the same issue. Trying to manually code sign it causes the error:
```
JRE.xcframework: bundle format unrecognized, invalid, or unsuitable
```

**Comment by dsanghan** [comment]
@tomball Any updates on this?

**Comment by tomball** [comment]
No, I don't have a personal certificate, don't have any access to a Google-wide one, and will be retiring soon. @litstrong (j2objc's project lead), do you have any update?

[One of you can contribute a fix](https://developers.google.com/j2objc/guides/how-to-contribute), as the `xcodebuild -create-xcframework` command is in the short [gen_xcframework.sh](https://github.com/google/j2objc/blob/master/scripts/gen_xcframework.sh) script. Since the certificate is personal and shouldn't be checked into any project, I suggest supporting a shell environment variable that defines it, then update the script to reference that variable.

**Comment by litstrong** [comment]
Ack, sorry for my late response.

Just start to pick this up, let me take a look and will respond by next week.

**Comment by litstrong** [comment]
Per Tom mentioned, the solution here is to update scripts ([gen_xcframework.sh](https://github.com/google/j2objc/blob/master/scripts/gen_xcframework.sh), etc.) to take personal certificate to bless the framework post Xcode15.

Given personal certificate is required, that means after Xcode15, we'll stop distributing frameworks but providing the above mentioned scripts.

@dsanghan Could you clarify "manually code sign it"? Did you run `codesign --timestamp -v --sign "<YOUR CERTIFICATE (ABCXYZ)>" "<YOUR.xcframework>"` and get the error?

**Comment by dsanghan** [comment]
@litstrong Yup I'm running `security find-identity -v -p codesigning` to get the correct certificate - then running `codesign --timestamp -v --sign "<cert id>" "JRE.xcframework"`

It returns `JRE.xcframework: bundle format unrecognized, invalid, or unsuitable`

**Comment by bryan1anderson** [comment]
Same issue as @dsanghan unfortunately. Xcode 15 just entered GM build phase, so this is going to be live all next week

**Comment by bryan1anderson** [comment]
"bundle format unrecognized, invalid, or unsuitable" is not one of the known errors in https://developer.apple.com/documentation/Xcode/verifying-the-origin-of-your-xcframeworks and I found someone suggesting folder structure might be to blame..

**Comment by bryan1anderson** [comment]
@mungler was able to get this fixed: https://github.com/google/j2objc/issues/2169#issuecomment-1719620864
@dsanghan 

The problem was indeed folder structure.

**Comment by dsanghan** [comment]
Super! Thanks @bryan1anderson

**Comment by mungler** [comment]
I'm preparing a PR with the fix now...

**Comment by mungler** [comment]
Ok, I have some hoops to jump through before im allowed to contribute a PR, but here is the relevant change: https://github.com/google/j2objc/pull/2174/files

**Comment by bryan1anderson** [comment]
@mungler thanks you were definitely the champ of my day. This is not an area of our project any of us like getting into and I made the stupid decision to plunge into Sonoma and couldn't use Xcode 14 as a result.

**Comment by mungler** [comment]
@bryan1anderson funnily enough, i've been wanting to upgrade too, but couldn't until we could build our app project 👍 

Glad it helped somebody else in the process :)

**Comment by dsanghan** [comment]
Thanks @mungler!

**Comment by mungler** [comment]
Fixed by PRs #2175 and #2181 

This issue should be closed now.

**Comment by mungler** [comment]
@litstrong FYI
